packages/devextreme/testing/testcafe/tests/dataGrid/focus/focusedRow/focusedRow.ts

```diff
@@ -876,3 +876,38 @@ test('Focused row should be shown after reloading the page (T1058983)', async (t
 }).after(async () => {
   await clearLocalStorage();
 });
+
+test('It is possible to focus row that was added via push method if previously row with same index was focused (T1202646)', async (t) => {
+  const dataGrid = new DataGrid('#container');
+
+  await t.click(dataGrid.getDataRow(0).element);
+
+  await ClientFunction(() => {
+    const grid = ($('#container') as any).dxDataGrid('instance');
+    grid.getDataSource().store().push([{
+      type: 'insert',
+      data: { value: 2 },
+    }]);
+  })();
```

Comment:
Question.

Maybe we should add this ClientFunction to the ```DataGrid``` testing model?
Something like method with the name ```pushDataToTheStore``` or something :)